projects/observability/src/pages/explorer/explorer.component.ts

```diff
@@ -118,11 +137,14 @@ export class ExplorerComponent {
   private static readonly VISUALIZATION_EXPANDED_PREFERENCE: string = 'explorer.visualizationExpanded';
   private static readonly RESULTS_EXPANDED_PREFERENCE: string = 'explorer.resultsExpanded';
   private readonly explorerDashboardBuilder: ExplorerDashboardBuilder;
+  private savedQueries: SavedQuery[] = [];
+  private currentContext: ExplorerGeneratedDashboardContext = ObservabilityTraceType.Api;
```

Comment:
I think we can make a do without having this `currentContext` variable here.
As to how, we can change the `contextChangeSubject` (not `currentContext$`)  from `Subject` to `BehaviorSubject`.
`BehaviorSubject` are `Subject` which directly give us their current value without being subscribed to. Given both of them are referring to type `ExplorerGeneratedDashboardContext`, we can do `contextChangeSubject.getValue()`. That will also facilitate setting the initial value :)

Reply:
Also, it's more of a nitpick comment, so you can ignore it if you want.

Reply:
I tried changing the Subject to BehaviourSubject by changing

```ts
  private readonly contextChangeSubject: Subject<ExplorerGeneratedDashboardContext> = new Subject();
```

to

```ts
  private readonly contextChangeSubject: BehaviorSubject<ExplorerGeneratedDashboardContext> = new BehaviorSubject(
    ObservabilityTraceType.Api
  );
```

But TypeScript is going crazy with errors:

<img width="590" alt="Screenshot 2022-05-16 at 2 02 40 PM" src="https://user-images.githubusercontent.com/29686866/168552274-282cf753-8380-48fc-9a4d-94f5341b5188.png">

Am I doing something wrong?

Reply:
Ohh, there are two ways to handle this, that would resolve everything. You can bake it in future PRs if it makes sense.

```typescript
private readonly contextChangeSubject: BehaviorSubject<ExplorerGeneratedDashboardContext> = new BehaviorSubject(ObservabilityTraceType.Api as ExplorerGeneratedDashboardContext);
```

OR

```typescript
private readonly contextChangeSubject: BehaviorSubject<ExplorerGeneratedDashboardContext> = new BehaviorSubject<ExplorerGeneratedDashboardContext>(ObservabilityTraceType.Api);
```

Reply:
That worked. Thanks!

Change added to the next PR in the below commit:

https://github.com/razorpay/hypertrace-ui/pull/35/commits/ae6eea6bb3c993482bc86f64e98bbc2ea6bd665a

But changing `Subject` to `BehaviorSubject` started failing an existing test, couldn't figure out why so I reverted the change.

Here's the failing test:

https://github.com/razorpay/hypertrace-ui/runs/6450485969?check_suite_focus=true#step:4:1235